text/0000-support-single-implementation.md

```diff
@@ -0,0 +1,131 @@
+
+### Supporting third-party needs
+
+Most third-party requirements should be addressable through either exporters or by extending the core API, depending on the particular requirement.
+
+For example, adding an extra ID to every span could be done by the exporter, whereas allowing the user to add attributes to spans must be done through the API itself.
+
+Exporters could possibly be extended to some sort of callback for relevant events (particularly span creation).
+
+### Migrating from the current architecture
+
+To the extent that the current architecture has already been implemented, a migration path may look something like:
+
+1. Start with the "minimal implementation"
+1. Send data to the particular event's exporter
+1. Make the "SDK" conform to the new exporter implementation
+1. Break out "transformer" logic from the SDK
+
+## Trade-offs and mitigations
+
+### Providing enough flexibility for vendors
+
+The biggest risk with this RFC is probably the question of whether it supports the needed flexibility.
+
+For potential future third-party needs that cannot be addressed with an exporter (or even a hook-supplemented exporter), the solution is likely to **add the functionality into the API** - particularly where multiple parties have the same or a similar need.
+
+See, for example, the [request for "transaction" support in OpenCensus](https://github.com/census-instrumentation/opencensus-specs/issues/229). A feature like this would be a strong candidate for getting added directly into the OpenTelemetry core.
+
+### Migration costs
+
+Given that OpenTelemetry is still very young and not yet production-ready, migration costs should be relatively small for both end users and implementors - particularly to the extent that the ambiguity between the API and SDK has slowed down implementation.
+
+## Prior art and alternatives
+
+### Prior art
+
+The three main prior art examples are OpenTracing, OpenCensus, and the existing OpenTelemetry architecture.
+
+**OpenTracing** handles export functionality by having each vendor (or other party) reimplement the whole API. While this maximizes the flexibility for vendors, it also means that everyone must redo a lot of the same logic - including any logic that may be considered vendor-agnostic.
+
+**OpenCensus** handles it by having a single implementation which then sends data to tracing and stats exporters. _(**Note:** While architecturally this is very similar to the RFC's proposal, key differences include OpenCensus itself being quite opinionated and sometimes stateful, depending on the implementation.)_
+
+The **existing OpenTelemetry architecture** essentially combines the OpenTracing and OpenCensus approaches. As with OpenTracing, the core API provides an interface that can then have multiple implementations. The equivalent to the OpenCensus "opinionated implementation" is then provided as a *particular* implementation (the SDK), which in turn supports exporters.
```

Comment:
If I understand correctly, the delta between this RFC and the OpenCensus approach is – essentially – that the scope of the "common implementation" proposed here is smaller than the one it OpenCensus?